vod/hls/m3u8_builder.c

```diff
@@ -810,6 +816,70 @@ m3u8_builder_append_index_url(
 	return p;
 }
 
+static size_t
+m3u8_builder_closed_captions_get_size(
+	media_set_t* media_set,
+	request_context_t* request_context)
+{
+	media_closed_captions_t* closed_captions;
+	size_t result = 0;
+	size_t base =
+		sizeof("\n\n") - 1 +
+		(sizeof(M3U8_EXT_MEDIA_BASE) - 1 + VOD_INT32_LEN +
+		sizeof(M3U8_EXT_MEDIA_TYPE_CLOSED_CAPTIONS) - 1 +
+		sizeof(M3U8_EXT_MEDIA_GROUP_ID_CLOSED_CAPTIONS) - 1 +
+		sizeof(M3U8_EXT_MEDIA_LANG) - 1 +
+		LANG_ISO639_3_LEN +
+		sizeof(M3U8_EXT_MEDIA_INSTREAM_ID) - 1 +
+		sizeof(M3U8_EXT_MEDIA_NON_DEFAULT) - 1 +
+		sizeof("\"\n") - 1);
```

Comment:
this doesn't seem aligned with the write code, e.g. I don't see any int32 here, and need to use M3U8_EXT_MEDIA_DEFAULT since it's slightly longer

Reply:
I modified this so that there is now an int32 as part of the group-id